### PR TITLE
keyIsBelow fix + tests

### PR DESCRIPTION
**Pull request by tom-wa**
rewrote keyIsBelow because there were still some cascading edge cases missing + tests

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you! Can you add more test cases with escaped slashes? In the implementation please use the next key level iteration, otherwise it certainly will get inconsistent over time. (Do not try to detect yourself if a `/` is escaped)

**Comment by tom-wa** [comment]
@markus2330 done

File: src/libs/elektra/keytest.c [hunk 1]
```diff
@@ -419,68 +449,68 @@ succeed_if (keyRel (key, check) >= 0, "should be the same or below");
...
+ user/key/myself
+ user/key/sibling
+ @endverbatim
  *
  * @code
-keySetName (key, "user/key/folder");
+ keySetName (key, "user/key/folder");
 keySetName (check, "user/notsame/folder");
 succeed_if (keyRel (key, check) < -2, "key is not below, but same namespace");
- * @endcode
```
Comment (markus2330) [inline src/libs/elektra/keytest.c]: there are a lot of formatting issues here! Did you use clang-format 3.8?

File: src/libs/elektra/keytest.c [hunk 2]
```diff
@@ -31,6 +31,8 @@
 #include "kdbinternal.h"
 #include "kdbprivate.h"
 
+#include <stdio.h>
```
Comment (markus2330) [inline src/libs/elektra/keytest.c]: This shouldn't be included here.

**Comment by markus2330** [comment]
Thank you, looks much nicer now. But there are bunch of weird formatting issues and you included stdio.h?

File: src/libs/elektra/keytest.c (hunk 1 above)
Comment (tom-wa) [inline src/libs/elektra/keytest.c]: i used scripts/reformate-source which seems to call clang-format-3.8 + i have clang-format-3.8 installed.
just ran the script again, same issue
